contracts/IndexerRegistry.sol

```diff
@@ -161,52 +158,57 @@ contract IndexerRegistry is Initializable, OwnableUpgradeable, Constants {
 
     /**
      * @notice Indexers call to update their Metadata.
-     * @param _metadata Indexer metadata to update
+     * @param metadata Indexer metadata to update
      */
-    function updateMetadata(bytes32 _metadata) external {
-        require(isIndexer[msg.sender], 'G002');
-        metadataByIndexer[msg.sender] = _metadata;
-        emit UpdateMetadata(msg.sender, _metadata);
+    function updateMetadata(bytes32 metadata) external {
+        require(indexers[msg.sender] != bytes32(0), 'G002');
+        require(metadata != bytes32(0), 'IR005');
+
+        indexers[msg.sender] = metadata;
+
+        emit UpdateMetadata(msg.sender, metadata);
     }
 
     /**
      * @notice Indexers call to set the controller account, since indexer only allowed to set one controller account, we need to remove the previous controller account.
-     * @param _controller The address of controller account, indexer to set
+     * @param controller The address of controller account, indexer to set
      */
-    function setControllerAccount(address _controller) external {
-        // ensure to not use a controller used by someone else
-        require(isIndexer[msg.sender], 'G002');
-        require(controllerToIndexer[_controller] == ZERO_ADDRESS, 'IR005');
-
-        // remove previous controller to indexer link
-        address prevController = indexerToController[msg.sender];
-        delete controllerToIndexer[prevController];
-
-        // add 2 directional links between indexer and controller
-        indexerToController[msg.sender] = _controller;
-        controllerToIndexer[_controller] = msg.sender;
-        emit SetControllerAccount(msg.sender, _controller);
+    function setControllerAccount(address controller) external {
+        require(indexers[msg.sender] != bytes32(0), 'G002');
```

Comment:
[suggestion] use internal function like `isIndexer` or modifier to replace the duplicate check `indexers[msg.sender] != bytes32(0)`

Reply:
Used modifier